src/js/internal/abort_listener.ts

```diff
@@ -0,0 +1,38 @@
+const { validateAbortSignal, validateFunction } = require("internal/validators");
+const { kResistStopPropagation } = require("internal/shared");
+
+const SymbolDispose = Symbol.dispose;
+
+/**
+ * @param {AbortSignal} signal
+ * @param {EventListener} listener
+ * @returns {Disposable}
+ */
+function addAbortListener(signal, listener) {
```

Comment:
Consider using type annotations instead of JSDoc types

Reply:
yep! i deleted that locally